qiskit/providers/provider.py

```diff
@@ -75,3 +75,108 @@ def __eq__(self, other):
         equal. Subclassed providers can override this behavior.
         """
         return type(self).__name__ == type(other).__name__
+
+
+class ProviderV2(Provider, ABC):
+    """Base class for a Backend Provider."""
+    version = 2
+
+    def get_backend(self, name=None, **kwargs):
```

Comment:
I think Jay wants to move away from getters and setters - e.g. use `provider.backned()` for a single backend and `provider.backends()` for multiple.